docs/internals/details.rst

```diff
@@ -0,0 +1,46 @@
+Implementation Details
+======================
+
+
+.. _log_density_propto:
+
+Speed of the ``propto`` argument
+--------------------------------
+
+The log density function provided by a Stan model has
+the ability drop additive constants from the calculation.
+This is indicated by the ``propto`` ("``prop``\ortional  ``to``")
+argument to function.
+
+If you are using an application such as an MCMC algorithm which requires
+gradients and only needs the log density up to a proportion, setting
+``propto=True`` will be at least as fast as setting ``propto=False``
+and is generally recommended (and the default value).
+
+However, in the case of the ``log_density`` function (which does not calculate
```

Comment:
I think it would be good to restate the "only needs the log density up to a proportion" bit again in this paragraph.